src/IPaymentToken.sol

```diff
@@ -0,0 +1,6 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity 0.8.19;
+
+interface IPaymentToken {
```

Comment:
We can't assume any specific implementation of transfer restrictions on external tokens. We can only check our tokens.